src/poetry/inspection/info.py

```diff
@@ -498,6 +499,7 @@ def from_directory(cls, path: Path, disable_build: bool = False) -> PackageInfo:
                     else:
                         info = get_pep517_metadata(path)
                 except PackageInfoError:
+                    info = cls.from_pyproject_toml(path)
```

Comment:
IMO, instead of putting this here, we should probably create a `from_project_static_files` which tries `from_pyproject_toml` and then `from_setup_files`, and then replace all existing `from_setup_files` usages with `from_project_static_files`.

Reply:
I'll have to do this on Monday when I'm back at my desk. Are you one of the maintainers of poetry? It says "FFY00 approved these changes 3 days ago."